sound/soc/sof/intel/hda.c

```diff
@@ -1173,7 +1173,10 @@ static int hda_sdw_machine_select(struct snd_sof_dev *sdev)
 			mach->mach_params.links = mach->links;
 			mach->mach_params.link_mask = mach->link_mask;
 			mach->mach_params.platform = dev_name(sdev->dev);
-			pdata->fw_filename = mach->sof_fw_filename;
+			if (mach->sof_fw_filename)
+				pdata->fw_filename = mach->sof_fw_filename;
+			else
+				pdata->fw_filename = pdata->desc->default_fw_filename;
```

Comment:
using a gcc extension this could be
```
pdata->fw_filename = mach->sof_fw_filename ? : pdata->desc->default_fw_filename;
```

Reply:
I think the sound maintainers esp Mark prefer the if/else